### ISSUE TITLE
OtherCode Single-Page Disassembly...

### ISSUE THREAD

**Issue by pobtastic** [skoolkid/skoolkit#136]
Hiya!

This isn't an issue as such, more just that I'm a bit confused about how to achieve something...

I can see [here](https://skoolkit.ca/docs/skoolkit/ref-files.html#othercode) that I'm able to add an "other code" section, which I've used in the past, and seen others use for loading routines (especially when they get overwritten), however ... for this question, I'm just outputting a bunch of source code remnants (and I'll let them know at https://tcrf.net/Category:ZX_Spectrum_games that this exists too). But I'm having a bit of trouble figuring out how to display it in the way I'd like to.

```
[Titles]
SourceCode-Index=Source Code Remnants

[OtherCode:SourceCode]

[Paths]
SourceCode-Index=source-code.html
SourceCode-CodePath=reference/source-codes.html
```

This works fine- but ... I understand I can output this as a single-page template-  which, I'm taking to mean - it has no index page?  As this is only one informational page, I guess this would be a bit nicer - but I've tried various things and I can't seem to work out how to do it.

As always, thank you so much for your work!

**Comment by skoolkid** [comment]
I'm not sure I understand what you're trying to do. Can you explain what you've tried (in addition to the ref file changes you've already described), what you expected, and what actually happened?

**Comment by pobtastic** [comment]
You know, after looking at the single-page template itself ... I think the only issue here is that I've not quite understood what a single-page template is... And it's not what I initially thought it was!

I think what I was expecting, was that rather than my main disassembly index page linking via OtherCode to a second index page, I'd thought that it would be able to link directly to the disassembly page.

So like: https://pobtastic.github.io/travelwithtrashman/

Clicking on "[Source Code Remnants](https://pobtastic.github.io/travelwithtrashman/source-code.html)" would take you directly to https://pobtastic.github.io/travelwithtrashman/reference/source-codes.html/7800.html ...

I think I got a bit confused, just as the ref page instructions said that the page would end up being CodeID/asm.html - and I'd just assumed that would mean I'd then not have an index page at all.  Which TBH, for what I'm trying to show here, isn't really necessary given there's only ever going to be one entry in the index.

Actually ... about that index page, another thing I'd noticed which was unexpected, I tried adding a `[MemoryMap:SourceCode-Index]` and used `Intro=Hello...` and it removed the map title and link (the "address/ description" table header remained though.  I've never used this before, so IDK if I was also supposed to add any other items here?

**Comment by skoolkid** [comment]
To create a link on the disassembly home page to a page that already exists (i.e. is already automatically generated), you can first add a [[Page]](https://skoolkit.ca/docs/skoolkit/ref-files.html#page) section with a `Content` parameter and nothing else, and then modify the relevant `[Index:*:*]` section accordingly. (For reference, see the 'Command lists' link on the home page of my Skool Daze disassembly.)

What are you trying to accomplish with the `[MemoryMap:SourceCode-Index]` section?

**Comment by pobtastic** [comment]
> What are you trying to accomplish with the [MemoryMap:SourceCode-Index] section?

For that, it was only to add an Intro on the index page - but I think I just don't want the index page in the menu (so I'll follow what you've written above).

Just out of interest, how do I set just the OtherCode section to be a single-page disassembly?  It's not massively important for this, just I tried a few things that I thought would work (and they didn't).
